### PR TITLE
chore: PR feedback

### PR DESCRIPTION
**Pull request by haveanicedavid**
Started writing PR feedback, but figured it was easier to just make a PR into yours

The main difference is that I'm using Zod instead of AJV for validation, because with Zod we can use the schema to define the type definition (though `z.infer` like we're using with many of the form values) so that schema operates as the source of truth for both validation + type safety

The diff is bigger than it would otherwise be because I renamed one of the `error` functions which wasn't super clear

### CODE REVIEW

File: src/api/group.utils.ts [hunk 1]
```diff
@@ -3,35 +3,20 @@ import type {
...
-    } catch (e) {
-      metadata = {
-        name: `Group #${group.id}`,
-        updatedAt: '',
-      }
-    }
-  } else {
-    metadata = {
+  return {
+    metadata: getGroupMetadata(group.metadata, {
```
Comment (haveanicedavid) [inline src/api/group.utils.ts:16]: part of the reason for extracting these into a separate function is that it'll be easier to integrate fetching off-chain metadata

File: src/util/validation/metadata.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+import { valid } from './zod'
+
+const groupMetadataSchema = z.object({
+  name: valid.name,
+  description: valid.description.optional(),
+  updatedAt: z.string(),
+  forumLink: valid.url.optional(),
+  other: z.string().optional(),
+})
+export type UIGroupMetadata = z.infer<typeof groupMetadataSchema>
```
Comment (haveanicedavid) [inline src/util/validation/metadata.ts:12]: type defined from schema

File: src/util/validation/metadata.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+function isGroupMetadata(metadata: unknown): metadata is UIGroupMetadata {
+  return groupMetadataSchema.safeParse(metadata).success
```
Comment (haveanicedavid) [inline src/util/validation/metadata.ts:15]: `parse` and `safeParse` are zod's rough equivalent to the AJV `validate` methods (... I think)

**Review by haveanicedavid — COMMENTED** [review]
LMK what you think @ryanchristo

File: src/api/proposal.utils.ts [hunk 4]
```diff
@@ -5,13 +5,12 @@ import type {
...
   ProposalStakeFormValues,
   UIProposal,
-  UIProposalMetadata,
   Vote,
   VoteSDKType,
 } from 'types'
 import { toDate } from 'util/date'
 import { throwError } from 'util/errors'
-import { validateGroupProposalMetadata } from 'util/metadata'
+import { getProposalMetadata } from 'util/validation/metadata'
```
Comment (ryanchristo) [inline src/api/proposal.utils.ts:13]:
```suggestion
import { getProposalMetadata } from 'util/validation'
```

**Review by ryanchristo — APPROVED** [review]
Looks good to me. Thanks for feedback pr. I used `ajv` because of the todos but looks like those might have been outdated and `zod` is the route forward. :+1:

File: src/api/group.utils.ts (hunk 1 above)
Comment (ryanchristo) [reply src/api/group.utils.ts:16]: make sense

**Pull request merged**